src/lsm/lsm_cursor.c

```diff
@@ -889,6 +889,79 @@ err:	__clsm_leave(clsm);
 }
 
 /*
+ * __clsm_next_random --
+ *	WT_CURSOR->next method for the LSM cursor type when configured with
+ * next_random.
+ */
+static int
+__clsm_next_random(WT_CURSOR *cursor)
+{
+	WT_CURSOR_LSM *clsm;
+	WT_CURSOR *c;
+	WT_DECL_RET;
+	WT_LSM_CHUNK *chunk;
+	WT_SESSION_IMPL *session;
+	uint64_t checked_docs, i, rand_doc, total_docs;
+	const char *cfg[3];
+
+	clsm = (WT_CURSOR_LSM *)cursor;
+
+	CURSOR_API_CALL(cursor, session, next, NULL);
+	WT_CURSOR_NOVALUE(cursor);
+	WT_ERR(__clsm_enter(clsm, false, false));
+
+	cfg[0] = WT_CONFIG_BASE(session, WT_SESSION_drop);
+	cfg[1] = "next_random=true";
+	cfg[2] = NULL;
+
+	/* Initialize to the 0th chunk */
+	chunk = clsm->lsm_tree->chunk[0];
```

Comment:
I'm not clear why this initialization is here - the following for loop doesn't seem to require it. It would be better if {{chunk}} had a more descriptive name - AFAI can see it is the randomly selected chunk we are going to search within.